gym/dev_wrappers/tf_wrapper.py

```diff
@@ -0,0 +1,48 @@
+from typing import Union
+
+import jax.dlpack
+import tensorflow as tf
+
+from gym import Env, Wrapper
+
+
+class TensorFlowWrapper(Wrapper):
+    def __init__(self, env: Union[Env, Wrapper]):
+
+        super().__init__(env)
+        self.is_vector_env = getattr(env, "is_vector_env", False)
+
+    def step(self, action: tf.Tensor):
+        jax_action = self._tf_to_jax(action)
+        obs, reward, done, info = super().step(jax_action)
+
+        if self.is_vector_env:
+            obs = [self._jax_to_tf(o) for o in obs]
+        else:
+            obs = self._jax_to_tf(obs)
+
+        return obs, reward, done, info
+
+    def reset(self, **kwargs) -> Union[tf.Tensor, tuple[tf.Tensor, dict]]:
+        return_info = kwargs.get("return_info", False)
+        if return_info:
+            obs, info = super().reset(**kwargs)
```

Comment:
Is it preferred to interact with the wrapped environment this way or with `self.env.reset(...)`? Are they functionally the same or is there ever a scenario where they will behave differently? The same goes for calls to `step`.

I noticed in the Brax repo for their wrappers they use this method, but in other wrappers within gym the `self.env` method is used.